lib/resource/Directory.js

```diff
@@ -75,7 +77,19 @@ Directory.prototype.getProvider = function getProvider(/* [options,] callback */
     return callback();
   }
 
-  return this.dataStore.getResource(this.provider.href, options, require('./Provider'), callback);
+  function providerFactory(provider) {
```

Comment:
https://gist.github.com/timothyej/74d9523da136406b9f77

The problem is that it doesn't recognise the `providerFactory` function as a *Resource constructor*. It must be a constructor of `Resource`. See `isAssignableFrom` in `utils.js`.